java-manta-client/src/main/java/com/joyent/manta/config/BaseChainedConfigContext.java

```diff
@@ -312,7 +312,7 @@ public String getEncryptionPrivateKeyPath() {
     public void overwriteWithContext(final ConfigContext context) {
         /* If a default context is being used to overwrite after this
          * context has been initialized, then we want to be careful to not
-         * overwrite values that have already been set with defaults. */
+         * overwrite values that have already been set by non-default contexts. */
         boolean isDefaultContext = context instanceof DefaultsConfigContext;
```

Comment:
While it's unlikely that someone would extend `DefaultsConfigContext` we could make that class `final`. We'd lose the ability to mock that class using plain Mockito but it would enforce the expectation that defaults are only coming from one place.